InventorySales/Plugin/CatalogInventory/Model/Quote/Item/QuantityValidator/Initializer/StockItemPlugin.php

```diff
@@ -0,0 +1,192 @@
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+
+/**
+ * Plugin for initializes quantity stock item validator
+ *
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class StockItemPlugin
+{
+    /**
+     * @var FormatInterface
+     */
+    private $format;
+
+    /**
+     * @var AreProductsSalableForRequestedQtyInterface
+     */
+    private $areProductsSalableForRequestedQty;
+
+    /**
+     * @var IsProductSalableForRequestedQtyRequestInterfaceFactory
+     */
+    private $isProductSalableForRequestedQtyRequestInterfaceFactory;
+
+    /**
+     * @var StockResolverInterface
+     */
+    private $stockResolver;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var BackOrderNotifyCustomerCondition
+     */
+    private $backOrderNotifyCustomerCondition;
+
+    /**
+     * @var GetStockItemConfigurationInterface
+     */
+    private $getStockItemConfiguration;
+
+    /**
+     * @param FormatInterface $format
+     * @param AreProductsSalableForRequestedQtyInterface $areProductsSalableForRequestedQty
+     * @param IsProductSalableForRequestedQtyRequestInterfaceFactory $isProductSalableForRequestedQtyRequestFactory
+     * @param StockResolverInterface $stockResolver
+     * @param StoreManagerInterface $storeManager
+     * @param BackOrderNotifyCustomerCondition $backOrderNotifyCustomerCondition
+     * @param GetStockItemConfigurationInterface $getStockItemConfiguration
+     * @SuppressWarnings(PHPMD.LongVariable)
+     */
+    public function __construct(
+        FormatInterface $format,
+        AreProductsSalableForRequestedQtyInterface $areProductsSalableForRequestedQty,
+        IsProductSalableForRequestedQtyRequestInterfaceFactory $isProductSalableForRequestedQtyRequestFactory,
+        StockResolverInterface $stockResolver,
+        StoreManagerInterface $storeManager,
+        BackOrderNotifyCustomerCondition $backOrderNotifyCustomerCondition,
+        GetStockItemConfigurationInterface $getStockItemConfiguration
+    ) {
+        $this->format = $format;
+        $this->areProductsSalableForRequestedQty = $areProductsSalableForRequestedQty;
+        $this->isProductSalableForRequestedQtyRequestInterfaceFactory = $isProductSalableForRequestedQtyRequestFactory;
+        $this->stockResolver = $stockResolver;
+        $this->storeManager = $storeManager;
+        $this->backOrderNotifyCustomerCondition = $backOrderNotifyCustomerCondition;
+        $this->getStockItemConfiguration = $getStockItemConfiguration;
+    }
+
+    /**
+     * Set backorder qty to Quote Item
+     *
+     * @param StockItem $subject
+     * @param DataObject $result
+     * @param StockItemInterface $stockItem
+     * @param Item $quoteItem
+     * @param int $qty
+     *
+     * @return DataObject
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterInitialize(
```

Comment:
By introducing this plugin, the performance of quote item initialization may be significantly decreased since added salability check

Reply:
Performance tests passed successfully